### ISSUE TITLE
RFC: choose a license

### ISSUE THREAD

**Issue by mcfiredrill** [datafruits/datafruits#430]
This is just an issue to try to get a discussion going around which license is best.

We have [one PR open to add the MIT license](https://github.com/datafruits/datafruits/pull/344).
Certainly having a license would be better than having none, but I'd like to explore other possibilities for a license.

Here are a couple different license I've been trying to learn more about.

## Hippocratic License
https://firstdonoharm.dev/

Basically states that the software cannot be used to violate any human rights or cause harm.

Recently was adopted in another project I've contributed to. https://github.com/vcr/vcr/blob/master/LICENSE

## Copyfarleft

I am still learning about this but the idea seems to be that the software cannot be used in any for profit organizations that aren't structured like a cooperative.

https://wiki.p2pfoundation.net/Copyfarleft#Copyfarleft_type_licenses

**Comment by dfridkin** [comment]
I think this is an interesting point regarding copyfarleft:

> A copyfarleft license should make it possible for producers to share freely and to retain the value of their labour product, in otherwords it must be possible for workers to make money by applying their own labour to mutual property, but impossible for owners of private property to make money using wage labour.

I think this could create an issue where people such as myself are legally disallowed from contributing, as we are providing labor (code) without receiving a share of the product's revenue (Patreon) proportional to the value of our contribution.

I think that could be avoided by registering datafruits as a 501(c)(3) organization, although honestly it could be argued that the contributors receive the benefit of their work simply by the work being put into production -- if I contribute a feature, and then the feature is on the site, am I not receiving the value that I contribute?

That being said, the [Peer Production License](https://wiki.p2pfoundation.net/Peer_Production_License) seems like a really good fit. Relevant sections copied below, TL;DR at the bottom.

> ### 2. FAIR DEALING RIGHTS
>
> Nothing in this license is intended to reduce, limit, or restrict any uses free from copyright or rights arising from limitations or exceptions that are provided for in connection with the copyright protection under copyright law or other applicable laws.

> ### 3. LICENSE GRANT
>
> Subject to the terms and conditions of this License, [datafruits] hereby grants You a worldwide, royalty-free, non-exclusive, perpetual (for the duration of the applicable copyright) license to exercise the rights in the Work as stated below:
>
>    a. to Reproduce the Work, to incorporate the Work into one or more Collections, and to Reproduce the Work as incorporated in the Collections;
>
>    b. to create and Reproduce Adaptations provided that any such Adaptation, including any translation in any medium, takes reasonable steps to clearly label, demarcate or otherwise identify that changes were made to the original Work [...]
>
>    c. to Distribute and Publicly Perform the Work including as incorporated in Collections; and,
>
>    d. to Distribute and Publicly Perform Adaptations. [...] all rights not expressly granted by Licensor are hereby reserved, including but not limited to the rights set forth in Section 4(f).

> ### 4. RESTRICTIONS
>
> The license granted in Section 3 above is expressly made subject to and limited by the following restrictions:
>
> a. You may distribute or publicly perform the Work only under the terms of this license. You must include a copy of, or the uniform resource identifier (uri) for, this license with every copy of the Work You distribute or publicly perform [...]
>
> b. Subject to the exception in Section 4(c), You may not exercise any of the rights granted to You in Section 3 above in any manner that is primarily intended for or directed to ward commercial advantage or private monetary compensation [...]
>
> c. You may exercise the rights granted in Section 3 for commercial purposes only if:
>
> i. You are a Worker-owned business or Worker-owned collective; and
>
> ii. All financial gain, surplus, profits and benefits produced by the business or collective are distributed among the Workerowners
>
> d. Any use by a business that is privately owned and managed, and that seeks to generate profit from the labor of employees paid by salary or other wages, is not permitted under this license.
>
> e. If You distribute, or publicly perform the Work or any Adaptations or Collections, You must, unless a request has been made pursuant to Section 4(a), keep intact all copyright notices for the Work and provide [...] (i) the name of the Original Author (or pseudonym, if applicable) if supplied [...]; (ii) the title of the Work if supplied; (iii) to the extent reason ably practicable, the uri, if any, that Licensor specifies to be associated with the Work [...]; and, (iv) consistent with Section 3(b), in the case of an adaptation, a credit identifying the use of the Work in the adaptation [...]

TL;DR: According to my understanding, this license basically says that this product is FOSS _unless_ it's for commercial gain, _unless_ that gain is by a worker-owner collective.

The only issue I can find is that it requires a copyright. You could just stick a © on the project for [an automated copyright](https://www.copyright.gov/help/faq/faq-general.html#automatic), although [there is a problem with enforcing it](https://fairuse.stanford.edu/overview/faqs/software/#why_the_automatic_copyright_isn8217t).

I also really like the hippocratic license, and honestly you could use both of them -- I know Rust uses two licenses, but I think that typically that's to provide a commercial alternative for GPL software, and I'm not sure if a dual-license means that _both_ licenses apply or _either_ license applies.

Let me know what you think!

**Comment by dfridkin** [comment]
Two more thoughts I had:

1. I bet we could write in the README that _both_ licenses apply, and take both.
2. It'd be worth using something like [go-license-detector](https://github.com/go-enry/go-license-detector) to make sure that we're not infringing on the terms of any of the packages we're using.

**Comment by dudethatbe** [comment]
I was unable to get the go-license-detector to analyze the entire node_modules directory. If I specified package at a time with the utility, then it would reveal the used licenses. Maybe you can point me in the right direction? Not too experienced with Go already so... 😅
I used an npm-based [license checker](https://github.com/davglass/license-checker) and used a regex on the included licenses (slow, but I have a better idea now). Here's the whole output if anyones interested
[datafruits_dependency_licenses.txt](https://github.com/datafruits/datafruits/files/4755987/datafruits_dependency_licenses.txt)

Here's all the licenses in use and any combinations if they exist:
- MIT
- Apache-2.0
- MIT OR Apache-2.0
- MIT OR [CC0-1.0](https://creativecommons.org/publicdomain/zero/1.0/deed.en)
- [BSD-3-Clause](https://opensource.org/licenses/BSD-3-Clause)
- [WTFPL](https://en.wikipedia.org/wiki/WTFPL)

It sounds like we all at least like hippocratic enough to maybe use this one first? I like the Peer Production License too, thanks for the breakdown @dfridkin . I'm still learning how licensing works and would have gone with WTFPL so I don't have to keep thinking about it 😁 At least it seems common enough for a repo to use an "either or" type of approach with multiple licenses (as opposed to joining the logic of both of them together).

**Comment by dfridkin** [comment]
I looked briefly through the dependencies' licenses, and i think we're fine to restrict the usage with either PPL or the Hippocratic license -- none of the licenses indicate that we're not allowed to do that, and Apache actually explicitly says we are. I think. 

That might have been BSD, actually. I can't remember. It was one of them.

Either way, my vote goes for both, but honestly it's not a huge deal to me and I'm fine with either.

**Comment by mcfiredrill** [comment]
This license seems cool....know of any projects using it ? https://thufie.lain.haus/NPL.html

**Comment by mcfiredrill** [comment]
I think I like this one.
https://git.pixie.town/thufie/CNPL

**Comment by mcfiredrill** [comment]
Also looks interesting but the details seem a bit specific to Europe:
https://coopcycle.org/en/software/#digital-common
